components/message_identifier_router.jsx

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+import {joinChannel} from 'mattermost-redux/actions/channels';
+import {getUser, getUserByUsername, getUserByEmail} from 'mattermost-redux/actions/users';
+
+import ChannelView from 'components/channel_view';
+import UserStore from 'stores/user_store.jsx';
+import ChannelStore from 'stores/channel_store.jsx';
+import TeamStore from 'stores/team_store.jsx';
+import {Constants} from 'utils/constants.jsx';
+import {openDirectChannelToUser} from 'actions/channel_actions.jsx';
+import * as GlobalActions from 'actions/global_actions.jsx';
+import * as Utils from 'utils/utils.jsx';
+import store from 'stores/redux_store.jsx';
+const dispatch = store.dispatch;
+const getState = store.getState;
+
+const LengthOfId = 26;
+const LengthOfGroupId = 40;
+const LengthOfUserIdPair = 54;
```

Comment:
Also minor, but can we use all capitals for constants like so `LENGTH_OF_ID`? Just to be consistent with the rest of the codebase